vaadin-spring/src/main/java/com/vaadin/flow/spring/VaadinConfigurationProperties.java

```diff
@@ -87,6 +88,152 @@ public void setEnable(boolean enable) {
 
     }
 
+    /**
+     * Turns application to work in production mode. Production mode disables
+     * most of the logged information that appears on the console because
+     * logging and other debug features can have a significant performance
+     * impact. Development mode JavaScript functions are not exported, push is
+     * given as a minified JavaScript file instead of full size and static
+     * resources are cached. See Deploying to Production for more information.
+     */
+    private Boolean productionMode;
```

Comment:
Is this class gives the IDE assistance ( instead of `additional-spring-configuration-metadata.json` ) ?

Reply:
This comment is not resolved.

And in fact I don't like the way how this done.
I still don't get whether this gives out of the box the IDE assistance .

Another problem again: this is hardcoded and doesn't allow to code evolve.
Every time when property name is changed in Flow or a new one is added this needs to be changed and no-one 
will remember to do it.

So such approach either requires a test which checks that all the properties are used here 
OR
the approach which I prefer is: _generate_ a file based on the Vaadin properties.

Both files can be generated : Java and json.
Then the question how to include them into the project: the project maven config needs to be changed to include them.
Perhaps it's easier to generate json file and include a resource into the jar.
But it's up to you.

Reply:
Well, the truth is that I understand and agree with your concerns, Denis.

They lead me to a complete redesign of the PR, though, so I will think a little about it and post a new design to the issue as soon as I feel it's good enough.

Then just write the code and push the PR again.